app/services/gmaps/markers_builder.rb

```diff
@@ -0,0 +1,91 @@
+# extracted from 'gmaps4rails' gem
+# link = https://github.com/apneadiving/Google-Maps-for-Rails/blob/master/lib/gmaps4rails/markers_builder.rb
+module Gmaps
+  # the to_gmaps4rails method accepts a block to customize:
+  # - infowindow
+  # - picture
+  # - shadow
+  # - title
+  # - sidebar
+  # - json
+  #
+  # This works this way:
+  #   @users = User.all
+  #   @json  = Gmaps::MarkersBuilder.generate @users do |user, marker|
+  #     marker.lat user.latitude
+  #     marker.lng user.longitude
+  #     marker.infowindow render_to_string(
+  #       :partial => "/users/my_template",
+  #       :locals => { :object => user}
+  #       ).gsub(/\n/, '').gsub(/"/, '\"')
+  #     marker.picture({
+  #       :url    => "http://www.blankdots.com/img/github-32x32.png",
+  #       :width  => "64",
+  #       :height => "64",
+  #       :scaledWidth => "32", # Scaled width is half of the retina resolution; optional
+  #       :scaledHeight => "32", # Scaled width is half of the retina resolution; optional
+  #       })
+  #     marker.title   "i'm the title"
+  #     marker.json({ :id => user.id })
+  #   end
+
+
+  class MarkersBuilder
+
+    def self.generate(collection, &block)
+      MarkersBuilder.new(collection).call(&block)
+    end
+
+    def initialize(collection)
+      @collection = Array(collection)
+    end
+
+    def call(&block)
+      @collection.map do |object|
+        MarkerBuilder.new(object).call(&block)
+      end
+    end
+
+    class MarkerBuilder
+
+      attr_reader :object, :hash
+      def initialize(object)
+        @object = object
+        @hash   = {}
+      end
+
+      def call(&block)
```

Comment:
@tansaku this file was extracted from Gmaps4rails gem, the only method I added was `generate` in class `MarkersBuilder` [source](https://github.com/AgileVentures/LocalSupport/pull/397/files#diff-044234d81e24db3a723cd82292b9758dR35) 
The `MarkerBuilder` instance accepts a block that's why it seems that naming the method to `call` here was chosen to reveal the intention of calling blocks.